src/constructors.lua

```diff
@@ -236,11 +238,13 @@ end
 constructors["subpt"] = function(typ, pt)
     -- [[DP]]print("CONS: ", typ, pt, aux)
     local cache = ptcache[typ]
-    if not cache[pt.id] then
-        cache[pt.id] = newpattern{
+    local res_pt = cache[pt.id]
+    if not res_pt then
+        res_pt = newpattern{
             pkind = typ,
             pattern = pt
         }
+        cache[pt.id] = res_pt
```

Comment:
You forgot to change `return cache[pt.id]` to `return res_pt` as you did above and below.

Reply:
Done